distributed/client.py

```diff
@@ -2620,14 +2618,19 @@ def _graph_to_futures(
                 if deps:
                     dependencies[k] = list(set(dependencies.get(k, ())) | deps)
 
+            # We send the graph where all WrappedKey has been unpacked
+            dsk = {k: v[0] for k, v in d.items() if k is not v[0]}
+
+            # The scheduler expect all keys to be strings
+            dsk = str_graph(dsk)
+
             if isinstance(retries, Number) and retries > 0:
-                retries = {k: retries for k in dsk3}
+                retries = {k: retries for k in dsk}
 
-            futures = {key: Future(key, self, inform=False) for key in keyset}
             self._send_to_scheduler(
                 {
                     "op": "update-graph",
-                    "tasks": valmap(dumps_task, dsk3),
+                    "tasks": valmap(dumps_task, dsk),
```

Comment:
This is a hazy memory, but there may have been some value to creating the Futures on the client before spending a lont time submitting the graph.  I don't recall why this was though.  

We can probably ignore this and keep with the implementation here for now, but I wanted to raise this as a possible concern in case the issue that I am sort of remembering comes back.

Reply:
I am seeing some CI timeouts like: https://github.com/dask/distributed/pull/4127/checks?check_run_id=1166256353
Can that have something to do with this?
Or is it unrelated?